Mathlib/Data/Sigma/Basic.lean

```diff
@@ -0,0 +1,257 @@
+/-
+Copyright (c) 2017 Johannes Hölzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl
+-/
+import Mathlib.Init.Function
+import Std.Tactic.Ext
+import Mathlib.Logic.Function.Basic
+
+/-!
+# Sigma types
+
+This file proves basic results about sigma types.
+
+A sigma type is a dependent pair type. Like `α × β` but where the type of the second component
+depends on the first component. More precisely, given `β : ι → Type*`, `Sigma β` is made of stuff
+which is of type `β i` for some `i : ι`, so the sigma type is a disjoint union of types.
+For example, the sum type `X ⊕ Y` can be emulated using a sigma type, by taking `ι` with
+exactly two elements. See `equiv.sum_equiv_sigma_bool` (remark: probably not yet ported
+to Lean 4).
```

Comment:
```suggestion
exactly two elements. See `equiv.sum_equiv_sigma_bool`.
```
I think there's no need to leave warning about stuff we plan to port in future. These will probably never get cleaned up afterwards.

Reply:
I have punted for `Equiv.sum_equiv_sigma_bool` but I still don't properly understand naming conventions yet.

Reply:
I've just looked up what mathlib3port says and have now punted for `Equiv.sumEquivSigmaBool`